e2e/cloudbuild.yaml

```diff
@@ -0,0 +1,200 @@
+
+## This Cloud Build script is used to run the automated e2e test workflow
+
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+  id: Cache cloud-sdk image
+  waitFor: ['-']
+  args: ['pull', 'google/cloud-sdk:latest']
+  timeout: 120s
+- name: 'alpine'
+  id: Set up env.mk
+  waitFor: ['-']
+  entrypoint: /bin/ash
+  args:
+  - '-c'
+  - |
+      set -o errexit
+      set -o xtrace
+      set 
+      apk add gettext
+      cat e2e/env.mk.template | \
+      envsubst > env.mk
+  env:
+  - 'PROJECT_ID=${PROJECT_ID}'
+  - 'BUILD_ID=${_BUILD_ID_SHORT}'
+  timeout: 60s
+- name: 'google/cloud-sdk:latest'
+  id: Create DNS managed zone
+  waitFor: ['Cache cloud-sdk image']
+  args:
+  - gcloud
+  - dns
+  - managed-zones
+  - create
+  - ${_MANAGED_ZONE_NAME}
+  - --description="Managed zone for e2e testing commit ${SHORT_SHA}"
+  - --dns-name=${_DNS_NAME}
+  - --visibility=public
+  timeout: 60s
+- name: 'google/cloud-sdk:latest'
+  id: Export zone records
+  waitFor: ['Create DNS managed zone', 'Cache cloud-sdk image']
+  args:
+  - gcloud
+  - dns
+  - record-sets
+  - export
+  - --zone=${_MANAGED_ZONE_NAME}
+  - --zone-file-format
+  - ${_MANAGED_ZONE_NAME}.zonefile
+- name: 'alpine'
+  id: Remove SOA record
+  waitFor: ['Export zone records']
+  entrypoint: /bin/ash
+  args:
+  - '-c'
+  - |
+    set -o errexit
+    set -o xtrace
+    grep -v SOA ${_MANAGED_ZONE_NAME}.zonefile > ${_MANAGED_ZONE_NAME}.zonefile.modified
+- name: 'google/cloud-sdk:latest'
+  id: Import NS records to top level
+  waitFor: ['Remove SOA record', 'Cache cloud-sdk image']
+  args:
+  - gcloud
+  - dns
+  - record-sets
+  - import
+  - --zone=${_TOP_LEVEL_ZONE_NAME}
+  - --zone-file-format
+  - ${_MANAGED_ZONE_NAME}.zonefile.modified
+- name: 'ubuntu:latest'
+  id: Setup Firebase config
+  waitFor: ['-']
+  entrypoint: /workspace/scripts/firebase-config-setup.sh
+  args:
+  - ${PROJECT_ID}
+  - ${_WEB_API_KEY}
+- name: 'google/cloud-sdk:latest'
+  id: make cluster
+  waitFor: ['Set up env.mk', 'Cache cloud-sdk image']
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    make cluster || touch e2e-failed
+- name: 'google/cloud-sdk:latest'
+  id: make build-infrastructure
+  waitFor: ['make cluster']
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    make build-infrastructure || touch e2e-failed
+- name: 'google/cloud-sdk:latest'
+  id: make build-backend
+  waitFor: ['make build-infrastructure']
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    make build-backend || touch e2e-failed
+- name: 'google/cloud-sdk:latest'
+  id: make build-webui
+  waitFor: ['make build-infrastructure']
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    make build-webui || touch e2e-failed
+- name: 'google/cloud-sdk:latest'
+  id: make build-userservice
+  waitFor: ['make build-infrastructure']
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    make build-userservice || touch e2e-failed
+- name: 'launcher.gcr.io/google/ubuntu1604'
+  id: Wait for application to be reachable
```

Comment:
Does this wait have a timeout?

Reply:
Yes, it only makes 10 attempts (with a 10-second wait in between attempts). However, in answering this question I realized that there is not a timeout for the `curl` command itself, which could hang indefinitely, so I will make a change to add that.